packages/eslint-plugin-pf-codemods/src/rules/v6/buttonMoveIconsIconProp/button-moveIcons-icon-prop.test.ts

```diff
@@ -0,0 +1,85 @@
+const ruleTester = require("../../ruletester");
+import * as rule from "./button-moveIcons-icon-prop";
+
+ruleTester.run("button-moveIcons-icon-prop", rule, {
+  valid: [
+    {
+      code: `<Button variant="plain"><span>Some icon</span></Button>`,
+    },
+    {
+      code: `import { Button } from '@patternfly/react-core'; <Button icon={<span>Some icon</span>} />`,
+    },
+  ],
+  invalid: [
+    {
+      code: `import { Button } from '@patternfly/react-core'; const icon = <span>Some icon</span>; <Button variant="plain">{icon}</Button>`,
+      output: `import { Button } from '@patternfly/react-core'; const icon = <span>Some icon</span>; <Button icon={icon} variant="plain"></Button>`,
+      errors: [
+        {
+          message: `Icons must now be passed to the \`icon\` prop of Button instead of as children. If you are passing anything other than an icon as children, ignore this rule when running fixes.`,
+          type: "JSXElement",
+        },
+      ],
+    },
```

Comment:
Assuming that all children will always be removed by the codemod (which I _think_ is the case when it meets the expected criteria for the codemod to run) I think it would be nice if we converted it to a self closing tag, but I wouldn't block over that.